src/controllers/recipe/recipe.statics.ts

```diff
@@ -50,3 +52,35 @@ export const findAllRecipesLikedByUser = async ({
   }).exec();
   return results;
 };
+
+export const deleteRecipeById = async ({
+  userId,
+  recipeId,
+}: {
+  userId: string;
+  recipeId: string;
+}): Promise<TDeleteRecipeByIdResult> => {
+  const recipe = await RecipeModel.findById(recipeId);
+  const user = await UserModel.findById(userId);
+  if (!recipe || !user) {
+    throw new Error("Could not find the recipe or user");
+  }
+  if (recipe.postedBy === userId) {
```

Comment:
Let's split this into separate if statements and throw an error message that is more specific - one for recipe not found and one for user not found